### PR TITLE
Fld fun factor

### PR DESCRIPTION
**Pull request by fieker**

### CODE REVIEW

File: src/NumFieldOrd/NfOrd/MaxOrd/Polygons.jl [hunk 1]
```diff
@@ -113,7 +113,7 @@ end
 
 function lower_convex_hull(points::Vector{Tuple{Int, Int}})
 
-  points = sortpoints(points)
+  @show points = sortpoints(points)
```
Comment (thofma) [inline src/NumFieldOrd/NfOrd/MaxOrd/Polygons.jl:116]:
```suggestion
  points = sortpoints(points)
```

**Comment by thofma** [comment]
Something is off:
```julia
julia> Hecke.assertions(true)

julia> Qx, x = QQ["x"];

julia> f = -x^2 + x - 1;

julia> K, a = NumberField(f, "a", cached = false);

julia> simplify(K, cached = false);
ERROR: AssertionError: $(Expr(:escape, :(isposdef(g))))
Stacktrace:
 [1] macro expansion
   @ ~/.julia/dev/Hecke/src/Assertions.jl:214 [inlined]
 [2] _lll_quad(M::NfOrd)
   @ Hecke ~/.julia/dev/Hecke/src/NumFieldOrd/NfOrd/LLL.jl:355
 [3] lll(M::NfOrd; prec::Int64)
   @ Hecke ~/.julia/dev/Hecke/src/NumFieldOrd/NfOrd/LLL.jl:212
 [4] macro expansion
   @ ~/.julia/dev/Hecke/src/Hecke.jl:595 [inlined]
 [5] simplify(K::AnticNumberField; canonical::Bool, cached::Bool, save_LLL_basis::Bool)
   @ Hecke ~/.julia/dev/Hecke/src/NumField/NfAbs/Simplify.jl:65
 [6] top-level scope
   @ REPL[20]:1

```

**Comment by fieker** [comment]
conjugate_quad is wrong dues to f being non-monic

**Comment by thofma** [comment]
Here is the failing picard group. With the change here it sometimes gives `2` instead of `1`.
```julia
  K = number_field(x^4 + 2*x^3 - 35*x^2 - 36*x + 5, "a", cached = false)[1]
  E = equation_order(K)
  @test order(picard_group(E)[1]) == 1
```

**Comment by fieker** [comment]
On Thu, Sep 16, 2021 at 12:13:39AM -0700, Tommy Hofmann wrote:
The class group is already returning 2 sometimes.
Then the picard group has no choice.

Why? still no idea

**Comment by fieker** [comment]
On Thu, Sep 16, 2021 at 12:13:39AM -0700, Tommy Hofmann wrote:
> Here is the failing picard group. With the change here it sometimes gives `2` instead of `1`.
> ```julia
>   K = number_field(x^4 + 2*x^3 - 35*x^2 - 36*x + 5, "a", cached = false)[1]
>   E = equation_order(K)
>   @test order(picard_group(E)[1]) == 1
> ```
Progress:

julia> Hecke.zeta_log_residue(maximal_order(K), 0.6931/2)
[+/- 0.399]

with Carlo's lower_convex_hull

julia> Hecke.zeta_log_residue(maximal_order(K), 0.6931/2)
[+/- 0.963]

with the correct one...

**Comment by thofma** [comment]
Looks like `prime_decomposition_type` is wrong, or a variant thereof.

**Comment by fieker** [comment]
On Thu, Sep 16, 2021 at 02:46:29AM -0700, Tommy Hofmann wrote:
> Looks like `prime_decomposition_type` is wrong, or a variant thereof.
julia> prime_decomposition_type(maximal_order(K), 3)
3-element Vector{Tuple{Int64, Int64}}:
 (2, 1)
 (2, 1)
 (2, 1)

 julia> prime_decomposition_type(maximal_order(K), 3)
 2-element Vector{Tuple{Int64, Int64}}:
  (2, 1)
  (2, 1)

julia> degree(K)
4

**Comment by fieker** [comment]
On Thu, Sep 16, 2021 at 02:46:29AM -0700, Tommy Hofmann wrote:
> Looks like `prime_decomposition_type` is wrong, or a variant thereof.
I think I have it
The "empty" filter call in line 925 was wrong, it appears to be
neccessary to only use negative slopes (no idea why)
but it solves the problem: the missing slope was always(?) horizontal

**Comment by thofma** [comment]
Can you add the prime decomposition type test somewhere?

**Pull request merged**